### PR TITLE
Mon 15671 instance nego 241109

### PR DESCRIPTION
**Pull request by bouda1**
## Description

New negociation

REFS: MON-15671

### CODE REVIEW

File: common/src/file.cc [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+  files.sort();
+
+  EVP_MD_CTX* mdctx = EVP_MD_CTX_new();
+  EVP_DigestInit_ex(mdctx, EVP_sha256(), nullptr);
+
+  for (auto& f : files) {
+    const std::string& fname = std::filesystem::relative(f, dir_path).string();
+    EVP_DigestUpdate(mdctx, fname.data(), fname.size());
+    std::string content = read_file_content(f);
```
Comment (jean-christophe81) [inline common/src/file.cc:71]: in case of exception, mdctx is not freed

File: common/src/file.cc [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  std::string retval;
+  retval.reserve(SHA256_DIGEST_LENGTH * 2);
+  auto digit = [](unsigned char d) -> char {
+    if (d < 10)
+      return '0' + d;
+    else
+      return 'a' + (d - 10);
+  };
+
+  for (auto h : hash) {
```
Comment (jean-christophe81) [inline common/src/file.cc:89]: size is not taken into account for loop end

File: broker/unified_sql/src/stream_sql.cc [hunk 3]
```diff
@@ -1680,12 +1683,291 @@ void stream::_process_pb_host_parent(const std::shared_ptr<io::data>& d) {
...
+              auto* b = _eh_bind->bind(0).get();
+              b->set_value_as_u64(0, h.host_id());
+              b->next_row();
+            }
+            SPDLOG_LOGGER_TRACE(
+                _logger_sql,
+                "Check if some statements are ready, eh_bind connections "
+                "count = {}",
+                _eh_bind->connections_count());
+            if (_eh_bind->ready(0)) {
```
Comment (jean-christophe81) [inline broker/unified_sql/src/stream_sql.cc:1813]: _eh_bind->size() > 0 ?

File: broker/unified_sql/src/stream_sql.cc [hunk 4]
```diff
@@ -1680,12 +1683,291 @@ void stream::_process_pb_host_parent(const std::shared_ptr<io::data>& d) {
...
+              b->set_value_as_u64(0, s.host_id());
+              b->set_value_as_u64(1, s.service_id());
+              b->next_row();
+            }
+            SPDLOG_LOGGER_TRACE(
+                _logger_sql,
+                "Check if some statements are ready, es_bind connections "
+                "count = {}",
+                _es_bind->connections_count());
+            if (_es_bind->ready(0)) {
```
Comment (jean-christophe81) [inline broker/unified_sql/src/stream_sql.cc:1836]: same

File: common/src/file.cc (hunk 2 above)
Comment (bouda1) [reply common/src/file.cc:100]: sizeof(hash) == 32 so this is exactly what the loop does, 32 iterations.
Or I did not understand what you say...

**Review by jean-christophe81 — APPROVED** [review]

**Pull request merged**